[Bug]: DescribeCollection RPC fails in DataNode failover

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: latest
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

DataNode and DataCoord crashed before collection dropped. In failover, DescribeCollection RPCs from DataNode to RootCoord fail and retry endlessly. 

RootCoord Log
```
[2022/03/02 15:10:43.938 +08:00] [DEBUG] [root_coord.go:1376] [DescribeCollection] [role=rootcoord] ["collection name"=] [msgID=0]
[2022/03/02 15:10:43.938 +08:00] [ERROR] [root_coord.go:1388] ["DescribeCollection failed"] [role=rootcoord] ["collection name"=] [msgID=0] [error="can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*Core).DescribeCollection\n\t/Users/zilliz/milvus/internal/rootcoord/root_coord.go:1388\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).DescribeCollection\n\t/Users/zilliz/milvus/internal/distributed/rootcoord/service.go:354\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler.func1\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:915\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/Users/zilliz/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:917\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"]
[2022/03/02 15:10:43.943 +08:00] [DEBUG] [root_coord.go:1376] [DescribeCollection] [role=rootcoord] ["collection name"=] [msgID=0]
[2022/03/02 15:10:43.943 +08:00] [ERROR] [root_coord.go:1388] ["DescribeCollection failed"] [role=rootcoord] ["collection name"=] [msgID=0] [error="can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*Core).DescribeCollection\n\t/Users/zilliz/milvus/internal/rootcoord/root_coord.go:1388\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).DescribeCollection\n\t/Users/zilliz/milvus/internal/distributed/rootcoord/service.go:354\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler.func1\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:915\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/Users/zilliz/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:917\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"]
[2022/03/02 15:11:03.957 +08:00] [DEBUG] [root_coord.go:1376] [DescribeCollection] [role=rootcoord] ["collection name"=] [msgID=0]
[2022/03/02 15:11:03.957 +08:00] [ERROR] [root_coord.go:1388] ["DescribeCollection failed"] [role=rootcoord] ["collection name"=] [msgID=0] [error="can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*Core).DescribeCollection\n\t/Users/zilliz/milvus/internal/rootcoord/root_coord.go:1388\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).DescribeCollection\n\t/Users/zilliz/milvus/internal/distributed/rootcoord/service.go:354\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler.func1\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:915\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/Users/zilliz/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:917\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"]
```

DataNode Log
```
2022/03/02 15:10:23.915 +08:00] [DEBUG] [flow_graph_manager.go:39] ["received Vchannel Info"] [vChannelName=by-dev-rootcoord-dml_0_431542760693501249v0] ["Unflushed Segment Number"=0] ["Flushed Segment Number"=1]
[2022/03/02 15:10:23.915 +08:00] [DEBUG] [flow_graph_manager.go:39] ["received Vchannel Info"] [vChannelName=by-dev-rootcoord-dml_1_431542760693501249v1] ["Unflushed Segment Number"=0] ["Flushed Segment Number"=1]
[2022/03/02 15:10:23.920 +08:00] [DEBUG] [minio_kv.go:96] ["MinioKV new MinioKV success."]
[2022/03/02 15:10:23.920 +08:00] [INFO] [data_sync_service.go:190] ["Recover Segment NumOfRows form checkpoints"] [InsertChannel=by-dev-rootcoord-dml_1_431542760693501249v1] [SegmentID=431542760798158850] [NumOfRows=1497]
[2022/03/02 15:10:23.920 +08:00] [DEBUG] [segment_replica.go:382] ["Add Flushed segment"] ["segment ID"=431542760798158850] ["collection ID"=431542760693501249] ["partition ID"=431542760693501250] ["channel name"=by-dev-rootcoord-dml_1_431542760693501249v1]
[2022/03/02 15:10:23.920 +08:00] [INFO] [segment_replica.go:419] ["statsBinlogs is empty"]
[2022/03/02 15:10:23.922 +08:00] [DEBUG] [minio_kv.go:96] ["MinioKV new MinioKV success."]
[2022/03/02 15:10:23.922 +08:00] [INFO] [data_sync_service.go:190] ["Recover Segment NumOfRows form checkpoints"] [InsertChannel=by-dev-rootcoord-dml_0_431542760693501249v0] [SegmentID=431542760798158849] [NumOfRows=1503]
[2022/03/02 15:10:23.922 +08:00] [DEBUG] [segment_replica.go:382] ["Add Flushed segment"] ["segment ID"=431542760798158849] ["collection ID"=431542760693501249] ["partition ID"=431542760693501250] ["channel name"=by-dev-rootcoord-dml_0_431542760693501249v0]
[2022/03/02 15:10:23.922 +08:00] [INFO] [segment_replica.go:419] ["statsBinlogs is empty"]
[2022/03/02 15:10:23.922 +08:00] [ERROR] [segment_replica.go:673] ["Grpc error"] [error="describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).getCollectionSchema\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:673\ngithub.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).initPKBloomFilter\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:421\ngithub.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).addFlushedSegment\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:402\ngithub.com/milvus-io/milvus/internal/datanode.(*dataSyncService).initNodes\n\t/Users/zilliz/milvus/internal/datanode/data_sync_service.go:195\ngithub.com/milvus-io/milvus/internal/datanode.newDataSyncService\n\t/Users/zilliz/milvus/internal/datanode/data_sync_service.go:90\ngithub.com/milvus-io/milvus/internal/datanode.(*flowgraphManager).addAndStart\n\t/Users/zilliz/milvus/internal/datanode/flow_graph_manager.go:58\ngithub.com/milvus-io/milvus/internal/datanode.(*DataNode).handlePutEvent\n\t/Users/zilliz/milvus/internal/datanode/data_node.go:373\ngithub.com/milvus-io/milvus/internal/datanode.(*channelEventManager).Run.func1\n\t/Users/zilliz/milvus/internal/datanode/event_manager.go:56"]
[2022/03/02 15:10:23.922 +08:00] [WARN] [flow_graph_manager.go:60] ["new data sync service fail"] [vChannelName=by-dev-rootcoord-dml_1_431542760693501249v1] [error="describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"]
[2022/03/02 15:10:23.922 +08:00] [ERROR] [segment_replica.go:673] ["Grpc error"] [error="describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).getCollectionSchema\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:673\ngithub.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).initPKBloomFilter\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:421\ngithub.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).addFlushedSegment\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:402\ngithub.com/milvus-io/milvus/internal/datanode.(*dataSyncService).initNodes\n\t/Users/zilliz/milvus/internal/datanode/data_sync_service.go:195\ngithub.com/milvus-io/milvus/internal/datanode.newDataSyncService\n\t/Users/zilliz/milvus/internal/datanode/data_sync_service.go:90\ngithub.com/milvus-io/milvus/internal/datanode.(*flowgraphManager).addAndStart\n\t/Users/zilliz/milvus/internal/datanode/flow_graph_manager.go:58\ngithub.com/milvus-io/milvus/internal/datanode.(*DataNode).handlePutEvent\n\t/Users/zilliz/milvus/internal/datanode/data_node.go:373\ngithub.com/milvus-io/milvus/internal/datanode.(*channelEventManager).Run.func1\n\t/Users/zilliz/milvus/internal/datanode/event_manager.go:56"]
[2022/03/02 15:10:23.922 +08:00] [WARN] [event_manager.go:58] ["handle put event fail, starting retry"] [vChanName=by-dev-rootcoord-dml_1_431542760693501249v1] ["retry interval"=20s] [error="fail to add and start flowgraph for vChanName: by-dev-rootcoord-dml_1_431542760693501249v1, err: describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"]
[2022/03/02 15:10:23.922 +08:00] [WARN] [flow_graph_manager.go:60] ["new data sync service fail"] [vChannelName=by-dev-rootcoord-dml_0_431542760693501249v0] [error="describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"]
[2022/03/02 15:10:23.922 +08:00] [WARN] [event_manager.go:58] ["handle put event fail, starting retry"] [vChanName=by-dev-rootcoord-dml_0_431542760693501249v0] ["retry interval"=20s] [error="fail to add and start flowgraph for vChanName: by-dev-rootcoord-dml_0_431542760693501249v0, err: describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"]
```


### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Anything else?

Extracted from #15608

--- a/internal/datanode/data_sync_service.go
+++ b/internal/datanode/data_sync_service.go
@@ -170,7 +170,8 @@ func (dsService *dataSyncService) initNodes(vchanInfo *datapb.VchannelInfo) erro
 				pos:     *us.GetDmlPosition(),
 			}
 		}
-		if err := dsService.replica.addNormalSegment(us.GetID(), us.CollectionID, us.PartitionID, us.GetInsertChannel(), us.GetNumOfRows(), us.Statslogs, cp); err != nil {
+		if err := dsService.replica.addNormalSegment(us.GetID(), us.CollectionID, us.PartitionID, us.GetInsertChannel(),
+			us.GetNumOfRows(), us.Statslogs, cp, vchanInfo.GetSeekPosition().GetTimestamp()); err != nil {
 			return err
 		}
 	}
@@ -192,8 +193,8 @@ func (dsService *dataSyncService) initNodes(vchanInfo *datapb.VchannelInfo) erro
 			zap.Int64("SegmentID", fs.GetID()),
 			zap.Int64("NumOfRows", fs.GetNumOfRows()),
 		)
-		if err := dsService.replica.addFlushedSegment(fs.GetID(), fs.CollectionID,
-			fs.PartitionID, fs.GetInsertChannel(), fs.GetNumOfRows(), fs.Statslogs); err != nil {
+		if err := dsService.replica.addFlushedSegment(fs.GetID(), fs.CollectionID, fs.PartitionID, fs.GetInsertChannel(),
+			fs.GetNumOfRows(), fs.Statslogs, vchanInfo.GetSeekPosition().GetTimestamp()); err != nil {
 			return err
 		}
 	}

--- a/internal/datanode/segment_replica.go
+++ b/internal/datanode/segment_replica.go
@@ -52,9 +52,9 @@ type Replica interface {
 
 	listAllSegmentIDs() []UniqueID
 	addNewSegment(segID, collID, partitionID UniqueID, channelName string, startPos, endPos *internalpb.MsgPosition) error
-	addNormalSegment(segID, collID, partitionID UniqueID, channelName string, numOfRows int64, statsBinlog []*datapb.FieldBinlog, cp *segmentCheckPoint) error
+	addNormalSegment(segID, collID, partitionID UniqueID, channelName string, numOfRows int64, statsBinlog []*datapb.FieldBinlog, cp *segmentCheckPoint, recoverTs Timestamp) error
 	filterSegments(channelName string, partitionID UniqueID) []*Segment
-	addFlushedSegment(segID, collID, partitionID UniqueID, channelName string, numOfRows int64, statsBinlog []*datapb.FieldBinlog) error
+	addFlushedSegment(segID, collID, partitionID UniqueID, channelName string, numOfRows int64, statsBinlog []*datapb.FieldBinlog, recoverTs Timestamp) error
 	listNewSegmentsStartPositions() []*datapb.SegmentStartPosition
 	listSegmentsCheckPoints() map[UniqueID]segmentCheckPoint
 	updateSegmentEndPosition(segID UniqueID, endPos *internalpb.MsgPosition)
@@ -322,7 +322,7 @@ func (replica *SegmentReplica) filterSegments(channelName string, partitionID Un
 
 // addNormalSegment adds a *NotNew* and *NotFlushed* segment. Before add, please make sure there's no
 // such segment by `hasSegment`
-func (replica *SegmentReplica) addNormalSegment(segID, collID, partitionID UniqueID, channelName string, numOfRows int64, statsBinlogs []*datapb.FieldBinlog, cp *segmentCheckPoint) error {
+func (replica *SegmentReplica) addNormalSegment(segID, collID, partitionID UniqueID, channelName string, numOfRows int64, statsBinlogs []*datapb.FieldBinlog, cp *segmentCheckPoint, recoverTs Timestamp) error {
 	if collID != replica.collectionID {
 		log.Warn("Mismatch collection",
 			zap.Int64("input ID", collID),
@@ -352,7 +352,7 @@ func (replica *SegmentReplica) addNormalSegment(segID, collID, partitionID Uniqu
 		seg.checkPoint = *cp
 		seg.endPos = &cp.pos
 	}
-	err := replica.initPKBloomFilter(seg, statsBinlogs)
+	err := replica.initPKBloomFilter(seg, statsBinlogs, recoverTs)
 	if err != nil {
 		return err
 	}
@@ -370,7 +370,7 @@ func (replica *SegmentReplica) addNormalSegment(segID, collID, partitionID Uniqu
 
 // addFlushedSegment adds a *Flushed* segment. Before add, please make sure there's no
 // such segment by `hasSegment`
-func (replica *SegmentReplica) addFlushedSegment(segID, collID, partitionID UniqueID, channelName string, numOfRows int64, statsBinlogs []*datapb.FieldBinlog) error {
+func (replica *SegmentReplica) addFlushedSegment(segID, collID, partitionID UniqueID, channelName string, numOfRows int64, statsBinlogs []*datapb.FieldBinlog, recoverTs Timestamp) error {
 
 	if collID != replica.collectionID {
 		log.Warn("Mismatch collection",
@@ -399,7 +399,7 @@ func (replica *SegmentReplica) addFlushedSegment(segID, collID, partitionID Uniq
 		maxPK:    math.MinInt64, // use min value represents no value
 	}
 
-	err := replica.initPKBloomFilter(seg, statsBinlogs)
+	err := replica.initPKBloomFilter(seg, statsBinlogs, recoverTs)
 	if err != nil {
 		return err
 	}
@@ -414,9 +414,9 @@ func (replica *SegmentReplica) addFlushedSegment(segID, collID, partitionID Uniq
 	return nil
 }
 
-func (replica *SegmentReplica) initPKBloomFilter(s *Segment, statsBinlogs []*datapb.FieldBinlog) error {
+func (replica *SegmentReplica) initPKBloomFilter(s *Segment, statsBinlogs []*datapb.FieldBinlog, ts Timestamp) error {
 	log.Info("begin to init pk bloom filter", zap.Int("stats bin logs", len(statsBinlogs)))
-	schema, err := replica.getCollectionSchema(s.collectionID, 0)
+	schema, err := replica.getCollectionSchema(s.collectionID, ts)
 	if err != nil {
 		return err
 	}

--- a/internal/datanode/segment_replica_test.go
+++ b/internal/datanode/segment_replica_test.go
@@ -340,7 +340,7 @@ func TestSegmentReplica_InterfaceMethod(t *testing.T) {
 				sr.minIOKV = &mockMinioKV{}
 				assert.Nil(t, err)
 				require.False(t, sr.hasSegment(test.inSegID, true))
-				err = sr.addNormalSegment(test.inSegID, test.inCollID, 1, "", 0, []*datapb.FieldBinlog{getSimpleFieldBinlog()}, &segmentCheckPoint{})
+				err = sr.addNormalSegment(test.inSegID, test.inCollID, 1, "", 0, []*datapb.FieldBinlog{getSimpleFieldBinlog()}, &segmentCheckPoint{}, 0)
 				if test.isValidCase {
 					assert.NoError(t, err)
 					assert.True(t, sr.hasSegment(test.inSegID, true))
@@ -361,7 +361,7 @@ func TestSegmentReplica_InterfaceMethod(t *testing.T) {
 		segID := int64(101)
 		require.False(t, sr.hasSegment(segID, true))
 		assert.NotPanics(t, func() {
-			err = sr.addNormalSegment(segID, 1, 10, "empty_dml_chan", 0, []*datapb.FieldBinlog{}, nil)
+			err = sr.addNormalSegment(segID, 1, 10, "empty_dml_chan", 0, []*datapb.FieldBinlog{}, nil, 0)
 			assert.NoError(t, err)
 		})
 	})
@@ -587,9 +587,9 @@ func TestSegmentReplica_InterfaceMethod(t *testing.T) {
 
 		cpPos := &internalpb.MsgPosition{ChannelName: "insert-01", Timestamp: Timestamp(10)}
 		cp := &segmentCheckPoint{int64(10), *cpPos}
-		err = sr.addNormalSegment(1, 1, 2, "insert-01", int64(10), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp)
+		err = sr.addNormalSegment(1, 1, 2, "insert-01", int64(10), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp, 0)
 		assert.NotNil(t, err)
-		err = sr.addFlushedSegment(1, 1, 2, "insert-01", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()})
+		err = sr.addFlushedSegment(1, 1, 2, "insert-01", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, 0)
 		assert.NotNil(t, err)
 	})
 
@@ -600,9 +600,9 @@ func TestSegmentReplica_InterfaceMethod(t *testing.T) {
 
 		cpPos := &internalpb.MsgPosition{ChannelName: "insert-01", Timestamp: Timestamp(10)}
 		cp := &segmentCheckPoint{int64(10), *cpPos}
-		err = sr.addNormalSegment(1, 1, 2, "insert-01", int64(10), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp)
+		err = sr.addNormalSegment(1, 1, 2, "insert-01", int64(10), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp, 0)
 		assert.NotNil(t, err)
-		err = sr.addFlushedSegment(1, 1, 2, "insert-01", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()})
+		err = sr.addFlushedSegment(1, 1, 2, "insert-01", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, 0)
 		assert.NotNil(t, err)
 	})
 
@@ -613,9 +613,9 @@ func TestSegmentReplica_InterfaceMethod(t *testing.T) {
 
 		cpPos := &internalpb.MsgPosition{ChannelName: "insert-01", Timestamp: Timestamp(10)}
 		cp := &segmentCheckPoint{int64(10), *cpPos}
-		err = sr.addNormalSegment(1, 1, 2, "insert-01", int64(10), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp)
+		err = sr.addNormalSegment(1, 1, 2, "insert-01", int64(10), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp, 0)
 		assert.NotNil(t, err)
-		err = sr.addFlushedSegment(1, 1, 2, "insert-01", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()})
+		err = sr.addFlushedSegment(1, 1, 2, "insert-01", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, 0)
 		assert.NotNil(t, err)
 	})
 
@@ -678,7 +678,7 @@ func TestInnerFunctionSegment(t *testing.T) {
 
 	cpPos := &internalpb.MsgPosition{ChannelName: "insert-01", Timestamp: Timestamp(10)}
 	cp := &segmentCheckPoint{int64(10), *cpPos}
-	err = replica.addNormalSegment(1, 1, 2, "insert-01", int64(10), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp)
+	err = replica.addNormalSegment(1, 1, 2, "insert-01", int64(10), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp, 0)
 	assert.NoError(t, err)
 	assert.True(t, replica.hasSegment(1, true))
 	assert.Equal(t, 1, len(replica.normalSegments))
@@ -695,7 +695,7 @@ func TestInnerFunctionSegment(t *testing.T) {
 	assert.False(t, seg.isNew.Load().(bool))
 	assert.False(t, seg.isFlushed.Load().(bool))
 
-	err = replica.addNormalSegment(1, 100000, 2, "invalid", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, &segmentCheckPoint{})
+	err = replica.addNormalSegment(1, 100000, 2, "invalid", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, &segmentCheckPoint{}, 0)
 	assert.Error(t, err)
 
 	replica.updateStatistics(1, 10)
@@ -730,7 +730,7 @@ func TestInnerFunctionSegment(t *testing.T) {
 	replica.updateSegmentCheckPoint(1)
 	assert.Equal(t, int64(20), replica.normalSegments[UniqueID(1)].checkPoint.numRows)
 
-	err = replica.addFlushedSegment(1, 1, 2, "insert-01", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()})
+	err = replica.addFlushedSegment(1, 1, 2, "insert-01", int64(0), []*datapb.FieldBinlog{getSimpleFieldBinlog()}, 0)
 	assert.Nil(t, err)
 
 	totalSegments := replica.filterSegments("insert-01", common.InvalidPartitionID)
@@ -776,7 +776,7 @@ func TestReplica_UpdatePKRange(t *testing.T) {
 
 	err = replica.addNewSegment(1, collID, partID, chanName, startPos, endPos)
 	assert.Nil(t, err)
-	err = replica.addNormalSegment(2, collID, partID, chanName, 100, []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp)
+	err = replica.addNormalSegment(2, collID, partID, chanName, 100, []*datapb.FieldBinlog{getSimpleFieldBinlog()}, cp, 0)
 	assert.Nil(t, err)
 
 	segNew := replica.newSegments[1]

